crates/node/builder/src/builder/mod.rs

```diff
@@ -564,7 +564,7 @@ where
         let engine_tree_config = TreeConfig::default()
             .with_persistence_threshold(builder.config.engine.persistence_threshold)
             .with_memory_block_buffer_target(builder.config.engine.memory_block_buffer_target)
-            .with_state_root_task(builder.config.engine.state_root_task_enabled)
+            .with_disable_state_root_task(builder.config.engine.legacy_state_root_task_enabled)
```

Comment:
can we make naming consistent here, just so that it's less confusing?
```suggestion
            .with_legacy_state_root(builder.config.engine.legacy_state_root_task_enabled)
```

Reply:
> can we make naming consistent here, just so that it's less confusing?

sure done here https://github.com/paradigmxyz/reth/pull/14371/commits/8caa3376c5d9aaacdb905b962941a29e006bd134